### PR TITLE
[bitnami/elasticsearch] Elasticsearch enable security

### PR DESCRIPTION
**Pull request by Mauraza**
**Description of the change**
Enable the security x-pack, add username and password to elasticsearch

**Benefits**

**Possible drawbacks**

**Applicable issues**

  - fixes #

**Additional information**

**Checklist** 

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/elasticsearch/templates/data-statefulset.yaml [hunk 1]
```diff
@@ -178,6 +183,14 @@ spec:
...
               subPath: elasticsearch.yml
             {{- end }}
+            {{- if .Values.keystore }}
+            - name: keystore
+              emptyDir: {}
+            {{- range .Values.keystore }}
+            - name: keystore-{{ .secretName }}
+            {{- end}}
+          secret: {{ toYaml . | nindent 12 }}
+        {{- end }}
```
Comment (miguelaeh) [inline bitnami/elasticsearch/templates/data-statefulset.yaml:193]: This seems to have a wrong indentation

File: bitnami/elasticsearch/templates/ingest-deploy.yaml [hunk 2]
```diff
@@ -94,6 +94,12 @@ spec:
               value: "yes"
             - name: ELASTICSEARCH_NODE_TYPE
               value: "ingest"
+            - name: ELASTICSEARCH_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "elasticsearch.secretName" . }}
+                  key: password
+
```
Comment (miguelaeh) [inline bitnami/elasticsearch/templates/ingest-deploy.yaml:102]: Delete extra line

File: bitnami/elasticsearch/templates/data-statefulset.yaml [hunk 3]
```diff
@@ -178,6 +183,14 @@ spec:
               name: "config"
               subPath: elasticsearch.yml
             {{- end }}
+            {{- if .Values.keystore }}
+            - name: keystore
+              emptyDir: {}
+            {{- range .Values.keystore }}
+            - name: keystore-{{ .secretName }}
+            {{- end}}
+          secret: {{ toYaml . | nindent 12 }}
```
Comment (miguelaeh) [inline bitnami/elasticsearch/templates/data-statefulset.yaml:192]: Could we use here the tplValues function?

File: bitnami/elasticsearch/templates/coordinating-deploy.yaml [hunk 4]
```diff
@@ -99,6 +99,12 @@ spec:
               value: "yes"
             - name: ELASTICSEARCH_NODE_TYPE
               value: "coordinating"
+            - name: ELASTICSEARCH_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "elasticsearch.secretName" . }}
+                  key: password
+
```
Comment (miguelaeh) [inline bitnami/elasticsearch/templates/coordinating-deploy.yaml:107]: Delete extra line

File: bitnami/elasticsearch/README.md [hunk 5]
```diff
@@ -436,6 +438,11 @@ Find more information about how to deal with common errors related to Bitnami’
 
 ## Upgrading
 
+### To 15.0.0
+
+This version or later are licensed under the Elastic License that is not currently accepted as a Open Source license by the Open Source Iniciative (OSI).
```
Comment (miguelaeh) [inline bitnami/elasticsearch/README.md:443]:
```suggestion
From this version onwards, the chart is licensed under the Elastic License that is not currently accepted as an Open Source license by the Open Source Initiative (OSI).
```

File: bitnami/elasticsearch/README.md [hunk 6]
```diff
@@ -436,6 +438,11 @@ Find more information about how to deal with common errors related to Bitnami’
...
+And include this version or later are including x-pack plugin installed by default.
```
Comment (miguelaeh) [inline bitnami/elasticsearch/README.md:444]:
```suggestion
Also, from now on, the Helm Chart will include the x-pack plugin installed by default.
```

File: bitnami/elasticsearch/Chart.yaml [hunk 7]
```diff
@@ -25,4 +25,4 @@ name: elasticsearch
 sources:
   - https://github.com/bitnami/bitnami-docker-elasticsearch
   - https://www.elastic.co/products/elasticsearch
-version: 14.5.3
+version: 15.0.0
```
Comment (miguelaeh) [inline bitnami/elasticsearch/Chart.yaml:28]: I guess the major bump is due to the license change, right?

**Review by miguelaeh — CHANGES_REQUESTED** [review]
Thank you for the PR

**Comment by maxisam** [comment]
any timeline on margining this?

File: bitnami/elasticsearch/templates/data-statefulset.yaml (hunk 3 above)
Comment (maxisam) [reply bitnami/elasticsearch/templates/data-statefulset.yaml:192]: Where is the keystore?

**Comment by Mauraza** [comment]
Hi @maxisam,

We are working on this, but I can give you an estimation.

**Comment by marcosbc** [comment]
It seems like we also may need to update the exporter URI: https://github.com/justwatchcom/elasticsearch_exporter

**Comment by Mauraza** [comment]
There will be another PR adding this functionality to the chart. We are still working on it.

**Pull request closed without merging**